src/mh-core.jl

```diff
@@ -41,8 +41,8 @@ used if `chain_type=Chains`.
 types are `chain_type=Chains` if `MCMCChains` is imported, or 
 `chain_type=StructArray` if `StructArrays` is imported.
 """
-struct MetropolisHastings{D} <: Metropolis
-    proposal::D
+mutable struct MetropolisHastings{D} <: MHSampler
```

Comment:
Does it have to be mutable?
```suggestion
struct MetropolisHastings{D} <: MHSampler
```